### PR TITLE
node-fetch: restore types (for v2), update version, allow `agent: false`

### PR DESCRIPTION
**Pull request by glasser**
As discussed with @sandersn on #55533, this PR restores previously-deleted types for `node-fetch@2`.  It updates the version from v2.5 to v2.6 and fixes an error in the types where `agent: false` was not allowed.

The individual commits in the PR have more detailed commit messages.

### CODE REVIEW

**Comment by glasser** [comment]
When I run `npm test node-fetch` locally, I get:

```
glasser@dsg-mbp 0 15:42:46 ~/Projects/Apollo/DefinitelyTyped glasser/restore-node-fetch-v2-and-fix-agent-false u+4 $ npm test node-fetch

> definitely-typed@0.0.3 test
> dtslint types "node-fetch"

Error: /Users/glasser/Projects/Apollo/DefinitelyTyped/node_modules/form-data/index.d.ts:1:1
ERROR: 1:1  no-outside-dependencies  File /Users/glasser/Projects/Apollo/DefinitelyTyped/node_modules/form-data/index.d.ts comes from a `node_modules` but is not declared in this type's `package.json`.  See: https://github.com/Microsoft/dtslint/blob/master/docs/no-outside-dependencies.md

    at testTypesVersion (/Users/glasser/Projects/Apollo/DefinitelyTyped/node_modules/@definitelytyped/dtslint/bin/src/index.js:189:15)
    at async runTests (/Users/glasser/Projects/Apollo/DefinitelyTyped/node_modules/@definitelytyped/dtslint/bin/src/index.js:165:13)
    at async main (/Users/glasser/Projects/Apollo/DefinitelyTyped/node_modules/@definitelytyped/dtslint/bin/src/index.js:77:9)
```

I'm not sure why: this PR restores `node-fetch/package.json` which has an explicit dependency on `form-data`...

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## frisby ([<kbd>unpkg</kbd>](https://unpkg.com/browse/frisby@latest/))
was missing the following properties:
1. delete
1. delete
## make-fetch-happen ([<kbd>unpkg</kbd>](https://unpkg.com/browse/make-fetch-happen@latest/))
was missing the following properties:
1. FetchError
1. Headers
1. Request
1. Response
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against af88e0067437ed6a626f375486b3b3fee528e71e
</p>

**Comment by glasser** [comment]
The two packages mentioned by @DangerBotOSS only have dependencies changed in this PR (a revert of the dependency change from #55533).

**Review by trevor-scheer — APPROVED** [review]
Reviewed changes post-revert.

I don't understand the versioning situation - should what currently exists be 2.6 or do your changes move us to 2.6? I also need to refamiliarize myself with how versioning is handled in DT (mostly, should we be changing the header manually or is that automated in some way?). I'm guessing you did the right thing, only asking because I don't actually know.

**Review by abernix — APPROVED** [review]

**Comment by abernix** [comment]
That's my understanding of the goal too.  LGTM

**Comment by glasser** [comment]
By the way, if it's helpful, I'd be happy to be a new maintainer on this project.

File: types/node-fetch/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        ],
+        "strict": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
```
Comment (nicolas377) [inline types/node-fetch/tsconfig.json:16]:
```suggestion
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "lib": [
            "es6"
        ],
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
    },
```

**Review by nicolas377 — COMMENTED** [review]
Hopefully this will make the bot happy with the tsconfig

**Comment by glasser** [comment]
@nicolas377 I'll take your word for it! I guess something about what needs to be in tsconfig.json changed in the last few months?

File: types/node-fetch/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+//                 Niklas Lindgren <https://github.com/nikcorg>
+//                 Vinay Bedre <https://github.com/vinaybedre>
+//                 Antonio Román <https://github.com/kyranet>
+//                 Andrew Leedham <https://github.com/AndrewLeedham>
+//                 Jason Li <https://github.com/JasonLi914>
+//                 Steve Faulkner <https://github.com/southpolesteve>
+//                 ExE Boss <https://github.com/ExE-Boss>
+//                 Alex Savin <https://github.com/alexandrusavin>
+//                 Alexis Tyler <https://github.com/OmgImAlexis>
+//                 Jakub Kisielewski <https://github.com/kbkk>
```
Comment (nikcorg) [inline types/node-fetch/index.d.ts:13]: @glasser don't forget to include yourself in the list of contributors.

```suggestion
// Definitions by: Torsten Werner <https://github.com/torstenwerner>
//                 Niklas Lindgren <https://github.com/nikcorg>
//                 Vinay Bedre <https://github.com/vinaybedre>
//                 Antonio Román <https://github.com/kyranet>
//                 Andrew Leedham <https://github.com/AndrewLeedham>
//                 Jason Li <https://github.com/JasonLi914>
//                 Steve Faulkner <https://github.com/southpolesteve>
//                 ExE Boss <https://github.com/ExE-Boss>
//                 Alex Savin <https://github.com/alexandrusavin>
//                 Alexis Tyler <https://github.com/OmgImAlexis>
//                 Jakub Kisielewski <https://github.com/kbkk>
//                 David Glasser <https://github.com/glasser>
```

**Comment by peterblazejewicz** [comment]
It looks ok, ready to go?

**Comment by glasser** [comment]
@peterblazejewicz Thanks, applied your change!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@glasser thanks!
Please stay put in case something gets broken, thx!

**Comment by glasser** [comment]
Cool! I'll do so tomorrow when I'm in more "work mode".

**Review by nikcorg — APPROVED** [review]

**Comment by glasser** [comment]
Ready to merge

**Comment by glasser** [comment]
Thanks everyone! One final question: I see that this published to the `latest` tag, which previously had v3.0.2.  Is that reasonable? (If you're using v3 you should just be using the types that have been bundled in the package itself since v3.0.0, so I think this is ok?)

**Comment by trevor-scheer** [comment]
It seems like the v3's just ought to be deprecated (it looks like the most recent `3.0.3` actually was)

**Comment by peterblazejewicz** [comment]
PR: #58804

**Pull request merged**